libsolutil/CommonData.cpp

```diff
@@ -64,15 +64,15 @@ string solidity::util::toHex(bytes const& _data, HexPrefix _prefix, HexCase _cas
 
 	// Mixed case will be handled inside the loop.
 	char const* chars = _case == HexCase::Upper ? upperHexChars : lowerHexChars;
-	int rix = _data.size() - 1;
+	size_t rix = _data.size() - 1;
 	for (uint8_t c: _data)
 	{
 		// switch hex case every four hexchars
 		if (_case == HexCase::Mixed)
 			chars = (rix-- & 2) == 0 ? lowerHexChars : upperHexChars;
 
-		ret[i++] = chars[(unsigned(c) / 16) & 0xf];
-		ret[i++] = chars[unsigned(c) & 0xf];
+		ret[i++] = chars[unsigned(c / 16) & 0xfu];
+		ret[i++] = chars[unsigned(c) & 0xfu];
```

Comment:
This would change behavior if `c` could be negative. Fortunately it can't, so why not simplify it even further to `c / 16 & 0xfu`?

Also, seeing that this uses bit operations anyway, I wonder why `>> 4` is not used instead of `/ 16`.

Reply:
I am not sure about why `>> 4` was not used. Also not sure if I should change it in this PR :)
But you are right for that one